src/level/level_state.rs

```diff
@@ -51,16 +50,22 @@ impl Level {
         let level_grid = LevelGrid::from_grid(grid, world);
         let max_x = level_grid.grid().len();
         let max_y = level_grid.grid()[0].len();
-
-        world.add_resource(level_grid);
-
-        let mut queue = world.write_resource::<TileUpdateQueue>();
-        for x in 0..max_x {
-            for y in 0..max_y {
-                // write every coordinate in the update list to update every tile's mesh ans material
-                queue.push((x as i32, y as i32));
+        {
+            let mut tiles = world.write_storage::<Tile>();
+            let mut transforms = world.write_storage::<Transform>();
+            let dict = world.read_resource::<Vec<([[Tile; 3]; 3], String)>>();
+
+            for x in 0..max_x {
+                for y in 0..max_y {
+                    let (classifier, rotation) =
+                        level_grid.determine_sprite_for(x as i32, y as i32, &dict, &mut tiles);
+                    let entity = level_grid.get(x as i32, y as i32).unwrap();
+                    level_grid.set_transform(x as i32, y as i32, rotation, &mut transforms);
+                    insert_from_world(entity, classifier, world);
```

Comment:
so there's this code which asks the level grid for sprite for a tile and then asks the level grid for the tile itselft and then sets the transform of the tile and then inserts the tile somewhere, and this code exists in two places, and both places are neither `tile.rs` nor `level_grid.rs`. i would change that :)

Reply:
i think so, yes